.github/workflows/sonarqube.yml

```
@@ -18,10 +18,25 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0  # Important for SonarQube to see history
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)